### DISCUSSION TITLE
Approve Request

### DISCUSSION THREAD

**Discussion by kienkiat** [WalletConnect/walletconnect-monorepo#1546] [Developer Support]
Hi for the connector approve call_request eg, "eth_sendTransaction",
I am able to get the parameter id, however what do I have to provide for the 'result' parameter?
connector.approveRequest({
  id: 1,
  result: "0x41791102999c339c844880b23950704cc43aa840f3739e365323cda4dfa89e7a"
});

**Comment by Jake-Mulhern** [comment]
```
        const txResponse = await AlchemyUtils.sendWCTransaction(txObject, pk);

        const web3wallet = await createWallet();
        
        const response = { id, jsonrpc: "2.0", result: txResponse }

        const finalResponse = await web3wallet.respondSessionRequest({ topic, response });
```

This is the newest way to respond to a request

**Comment by Jake-Mulhern** [comment]
The docs are up to date now too
https://docs.walletconnect.com/2.0/javascript/web3wallet/wallet-usage

**Comment by AndonMitev** [comment]
Hey @Jake-Mulhern is there any changes in `"@walletconnect/client": "1.8.0",` as well? :(

I manage successfully to initialize session, how ever when i try to respond to any request like this for example:

```
connector.approveRequest({
              id: payload.id,
              result: signature,
            });
```

I feel like DAPP is not receiving the response because nothing is happening, and i'm referring more specific to `personal_sign` method, this is what dapp send to me as a wallet
